commit 5bd112ef893bd719f5b056a7328a6580f140d58a (refs/changes/54/3054/3)
Author: Rob Johnston <rob.johnston@joyent.com>
Date:   2017-12-08T01:09:05+00:00 (1 year, 10 months ago)
    
    OS-6496 Add platform-specific topo maps for the Joyent J310x platform

diff --git a/manifest b/manifest
index 6c1599d05f..ee7f0dbdc3 100644
--- a/manifest
+++ b/manifest
@@ -10204,7 +10204,11 @@ f usr/platform/i86pc/lib/fm/eft/intel.eft 0444 root sys
 d usr/platform/i86pc/lib/fm/topo 0755 root sys
 d usr/platform/i86pc/lib/fm/topo/maps 0755 root sys
 f usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-1101-disk-hc-topology.xml 0444 root sys
-f usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-3101-disk-hc-topology.xml 0444 root sys
+f usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-3101-hc-topology.xml 0444 root sys
+f usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-3102-hc-topology.xml 0444 root sys
+f usr/platform/i86pc/lib/fm/topo/maps/SSG-2028R-ACR24L-hc-topology.xml 0444 root sys
+f usr/platform/i86pc/lib/fm/topo/maps/SSG-2028R-ACR24L-chassis-hc-topology.xml 0444 root sys 
+f usr/platform/i86pc/lib/fm/topo/maps/SSG-2028R-ACR24L-disk-hc-topology.xml 0444 root sys 
 f usr/platform/i86pc/lib/fm/topo/maps/chassis-hc-topology.xml 0444 root sys
 f usr/platform/i86pc/lib/fm/topo/maps/chip-hc-topology.xml 0444 root sys
 f usr/platform/i86pc/lib/fm/topo/maps/fan-hc-topology.xml 0444 root sys
diff --git a/usr/src/lib/fm/topo/maps/Makefile b/usr/src/lib/fm/topo/maps/Makefile
index 9b235ab29a..7cf2a33dbe 100644
--- a/usr/src/lib/fm/topo/maps/Makefile
+++ b/usr/src/lib/fm/topo/maps/Makefile
@@ -51,7 +51,7 @@ i386_SUBDIRS = i86pc \
 		SUNW,Sun-Fire-X4540 \
 		SUNW,Sun-Fire-X4600-M2 \
 		Joyent,Joyent-Compute-Platform-1101 \
-		Joyent,Joyent-Compute-Platform-3101
+		SMCI,SSG-2028R-ACR24L
 
 SUBDIRS = $($(MACH)_SUBDIRS)
 
diff --git a/usr/src/lib/fm/topo/maps/Joyent,Joyent-Compute-Platform-3101/Makefile b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/Makefile
similarity index 64%
rename from usr/src/lib/fm/topo/maps/Joyent,Joyent-Compute-Platform-3101/Makefile
rename to usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/Makefile
index 9027d5aee9..1f19cdd4fa 100644
--- a/usr/src/lib/fm/topo/maps/Joyent,Joyent-Compute-Platform-3101/Makefile
+++ b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/Makefile
@@ -32,12 +32,29 @@
 ARCH = i86pc
 CLASS = arch
 DTDFILE = topology.dtd.1
-TOPOFILE = Joyent-Compute-Platform-3101-disk-hc-topology.xml
-SRCDIR = ../Joyent,Joyent-Compute-Platform-3101
+JOYENT_PLATFORMS = \
+	Joyent-Compute-Platform-3101-hc-topology.xml \
+	Joyent-Compute-Platform-3102-hc-topology.xml
 
-PLATFORM = Joyent-Compute-Platform-3101
-TOPOBASE = ../i86pc/i86pc-hc-topology.xml
+TOPOFILE = \
+	SSG-2028R-ACR24L-hc-topology.xml \
+	SSG-2028R-ACR24L-chassis-hc-topology.xml \
+	SSG-2028R-ACR24L-disk-hc-topology.xml
 
-CLOBBERFILES +=	$(TOPOFILE)
+TOPOFILE += $(JOYENT_PLATFORMS)
+
+SRCDIR = ../SMCI,SSG-2028R-ACR24L
+
+PLATFORM = SSG-2028R-ACR24L
+
+CLOBBERFILES +=	$(JOYENT_PLATFORMS) SSG-2028R-ACR24L-disk-hc-topology.xml
+
+#
+# Note, the Joyent Compute Platform 310x is based on the SuperMicro
+# SSG-2028R-ACR24L. Because of that, the topo map used here will work for all
+# such systems.
+# 
+$(JOYENT_PLATFORMS): SSG-2028R-ACR24L-hc-topology.xml
+	@ln -s $? $@
 
 include ../Makefile.map
diff --git a/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-chassis-hc-topology.xml b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-chassis-hc-topology.xml
new file mode 100644
index 0000000000..8a53632472
--- /dev/null
+++ b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-chassis-hc-topology.xml
@@ -0,0 +1,42 @@
+<?xml version="1.0"?>
+<!DOCTYPE topology SYSTEM "/usr/share/lib/xml/dtd/topology.dtd.1">
+<!--
+ Copyright (c) 2017, Joyent, Inc.
+
+
+ This file and its contents are supplied under the terms of the
+ Common Development and Distribution License ("CDDL"), version 1.0.
+ You may only use this file in accordance with the terms of version
+ 1.0 of the CDDL.
+
+ A full copy of the text of the CDDL should have accompanied this
+ source.  A copy of the CDDL is also available via the Internet at
+ http://www.illumos.org/license/CDDL.
+
+-->
+
+<topology name='chassis' scheme='hc'>
+  <range name='chassis' min='0' max='0'>
+    <node instance='0'>
+      <fac-enum provider='fac_prov_ipmi' />
+      <!--
+        chassis locate LED
+      -->
+      <facility name='locate' type='indicator' provider='fac_prov_ipmi' >
+          <propgroup name='facility' version='1' name-stability='Private'
+	     data-stability='Private' >
+                <propval name='type' type='uint32' value='1' />
+                <propmethod name='chassis_ident_mode' version='0'
+                    propname='mode' proptype='uint32' mutable='1' >
+                </propmethod>
+          </propgroup>
+      </facility>
+      <propgroup name='ipmi' version='1'
+          name-stability='Private' data-stability='Private' >
+         <propval name='entity-list' type='string_array' >
+           <propitem value='Chassis Intru' />
+	 </propval>
+      </propgroup>
+    </node>
+  </range>
+</topology>
diff --git a/usr/src/lib/fm/topo/maps/Joyent,Joyent-Compute-Platform-3101/Joyent-Compute-Platform-3101-disk-hc-topology.xmlgenksh b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-disk-hc-topology.xmlgenksh
similarity index 100%
rename from usr/src/lib/fm/topo/maps/Joyent,Joyent-Compute-Platform-3101/Joyent-Compute-Platform-3101-disk-hc-topology.xmlgenksh
rename to usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-disk-hc-topology.xmlgenksh
diff --git a/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-hc-topology.xml b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-hc-topology.xml
new file mode 100644
index 0000000000..3ec3018864
--- /dev/null
+++ b/usr/src/lib/fm/topo/maps/SMCI,SSG-2028R-ACR24L/SSG-2028R-ACR24L-hc-topology.xml
@@ -0,0 +1,90 @@
+<?xml version="1.0"?>
+<!DOCTYPE topology SYSTEM "/usr/share/lib/xml/dtd/topology.dtd.1">
+<!--
+Copyright (c) 2017, Joyent, Inc.
+
+ CDDL HEADER START
+
+ The contents of this file are subject to the terms of the
+ Common Development and Distribution License (the "License").
+ You may not use this file except in compliance with the License.
+
+ You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ or http://www.opensolaris.org/os/licensing.
+ See the License for the specific language governing permissions
+ and limitations under the License.
+
+ When distributing Covered Code, include this CDDL HEADER in each
+ file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ If applicable, add the following below this CDDL HEADER, with the
+ fields enclosed by brackets "[]" replaced with your own identifying
+ information: Portions Copyright [yyyy] [name of copyright owner]
+
+ CDDL HEADER END
+
+-->
+
+<topology name='i86pc' scheme='hc'>
+
+  <range name='motherboard' min='0' max='0'>
+    <node instance='0'>
+        <fac-enum provider='fac_prov_ipmi' />
+        <propgroup name='protocol' version='1'
+            name-stability='Private' data-stability='Private' >
+            <propval name='label' type='string' value='MB' />
+        </propgroup>
+        <propgroup name='ipmi' version='1'
+            name-stability='Private' data-stability='Private' >
+            <propval name='entity-list' type='string_array' >
+	        <propitem value='MB_10G Temp' />
+	        <propitem value='PCH Temp' />
+	        <propitem value='Peripheral Temp' />
+	        <propitem value='5VSB' />
+	        <propitem value='5VSB' />
+	        <propitem value='3.3VSB' />
+	        <propitem value='3.3VCC' />
+	        <propitem value='1.5V PCH' />
+	        <propitem value='1.2V BMC' />
+	        <propitem value='1.05V PCH' />
+	        <propitem value='12V' />
+	        <propitem value='5VCC' />
+	        <propitem value='VBAT' />
+	        <propitem value='VDIMMAB' />
+	        <propitem value='VDIMMCD' />
+	        <propitem value='VDIMMEF' />
+	        <propitem value='VDIMMGH' />
+	    </propval>
+        </propgroup>
+    </node>
+
+    <dependents grouping='children'>
+      <range name='chip' min='0' max='100'>
+        <enum-method name='chip' version='1' />
+      </range>
+      <range name='hostbridge' min='0' max='254'>
+        <enum-method name='hostbridge' version='1' />
+      </range>
+    </dependents>
+
+  </range>
+
+  <range name='chassis' min='0' max='0'>
+    <propmap name='SSG-2028R-ACR24L-chassis' />
+
+    <dependents grouping='children'>
+
+       <range name='psu' min='0' max='1'>
+          <enum-method name='ipmi' version='1' />
+       </range>
+       <range name='fan' min='0' max='100'>
+          <enum-method name='ipmi' version='1' />
+       </range>
+       <range name='bay' min='0' max='23'>
+          <propmap name='SSG-2028R-ACR24L-disk' />
+       </range>
+
+    </dependents>
+
+  </range>
+
+</topology>
diff --git a/usr/src/lib/fm/topo/maps/i86pc/i86pc-legacy-hc-topology.xml b/usr/src/lib/fm/topo/maps/i86pc/i86pc-legacy-hc-topology.xml
index 5bddd36cc2..1a459d854b 100644
--- a/usr/src/lib/fm/topo/maps/i86pc/i86pc-legacy-hc-topology.xml
+++ b/usr/src/lib/fm/topo/maps/i86pc/i86pc-legacy-hc-topology.xml
@@ -164,23 +164,6 @@ Copyright (c) 2017, Joyent, Inc.
        </range>
     </set>
 
-    <!--
-     Note, the Joyent Compute Platform 310x is based on the SuperMicro
-     SSG-2028R-ACR24L. Because of that, the topo map used here will work
-     for all such systems. Hence why it is included in the setlist.
-    -->
-    <set type='product' setlist='Joyent-Compute-Platform-3101|Joyent-Compute-Platform-3102|SSG-2028R-ACR24L'>
-       <range name='psu' min='0' max='1'>
-          <enum-method name='ipmi' version='1' />
-       </range>
-       <range name='fan' min='0' max='100'>
-          <enum-method name='ipmi' version='1' />
-       </range>
-       <range name='bay' min='0' max='23'>
-          <propmap name='Joyent-Compute-Platform-3101-disk' />
-       </range>
-    </set>
-
     <set type='product' setlist='default'>
        <range name='psu' min='0' max='100'>
    	  <enum-method name='ipmi' version='1' />
diff --git a/usr/src/pkg/manifests/service-fault-management.mf b/usr/src/pkg/manifests/service-fault-management.mf
index e83432f48f..6baf96a93d 100644
--- a/usr/src/pkg/manifests/service-fault-management.mf
+++ b/usr/src/pkg/manifests/service-fault-management.mf
@@ -21,6 +21,7 @@
 
 #
 # Copyright (c) 2010, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -730,7 +731,19 @@ $(i386_ONLY)file path=usr/platform/i86pc/lib/fm/eft/gcpu.eft mode=0444
 $(i386_ONLY)file path=usr/platform/i86pc/lib/fm/eft/gcpu_amd.eft mode=0444
 $(i386_ONLY)file path=usr/platform/i86pc/lib/fm/eft/intel.eft mode=0444
 $(i386_ONLY)file \
-    path=usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-1101-disk-hc-topology.xml \
+    path=usr/platform/i86pc/lib/fm/topo/maps/SSG-2028R-ACR24L-hc-topology.xml \
+    mode=0444
+$(i386_ONLY)file \
+    path=usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-3101-hc-topology.xml \
+    mode=0444
+$(i386_ONLY)file \
+    path=usr/platform/i86pc/lib/fm/topo/maps/Joyent-Compute-Platform-3102-hc-topology.xml \
+    mode=0444
+$(i386_ONLY)file \
+    path=usr/platform/i86pc/lib/fm/topo/maps/SSG-2028R-ACR24L-chassis-hc-topology.xml \
+    mode=0444
+$(i386_ONLY)file \
+    path=usr/platform/i86pc/lib/fm/topo/maps/SSG-2028R-ACR24L-disk-hc-topology.xml \
     mode=0444
 $(i386_ONLY)file \
     path=usr/platform/i86pc/lib/fm/topo/maps/Netra-X4200-M2-disk-hc-topology.xml \
